lib/galaxy/celery/tasks.py

```diff
@@ -0,0 +1,40 @@
+import sys
+
+from galaxy.celery import celery_app
+from galaxy.datatypes.registry import Registry
+from galaxy.model import (set_datatypes_registry, User)
+from galaxy.model.mapping import init
+from galaxy.model.orm.scripts import get_config
+from galaxy.util.custom_logging import get_logger
+
+log = get_logger(__name__)
+
+
+def get_galaxy_context():
+    # This insanity is because get_config (I think) is not cooperating with celery arg handling
+    _preserved_argv = list(sys.argv[1:])
+    del sys.argv[1:]
+
+    registry = Registry()
```

Comment:
@jmchilton I was hoping to ask you about this.  With the new DI stuff in mind, what's the right approach for acquiring the most minimal possible galaxy application context here (so, model, managers, config, security, and that's about it -- no threads)?

Reply:
I think it is fine to just just create the database model mapping object - seems totally fine for a first pass. We do that in the tests and stuff. If you wanted a bit more I might try to to define an interface and extract UniverseApplication into a smaller superclass.

Here is a rough pass at that, I haven't run it because my laptop is currently dying... all the time so I'm sure it doesn't work at all really but just working through some ideas:

https://github.com/galaxyproject/galaxy/commit/3ba7ae24ead169e787dbd00138249d8261fa76d1

I think the biggest issue here is that each task is going to need a different sized app. In theory DI is perfect for that but it might be awkward to get to that point where dependencies are declared and injected as needed for each task. We need to have the components themselves not depend on ``app``, we need to figure out how to order shutdown tasks, etc..

Sorry I don't have a more complete answer given my endless stream of technical issues.

Reply:
@jmchilton No worries at all, thanks so much!  Marius took your tweak and ran with it and it's working great.